Samples/Adventure/AdventureSocketClient/Program.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Orleans;
+using Orleans.Runtime.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using DotNetty.Codecs;
+using DotNetty.Transport.Bootstrapping;
+using DotNetty.Transport.Channels;
+using DotNetty.Transport.Channels.Sockets;
+
+
+namespace AdventureSocketClient
+{
+    class Program
+    {
+
+        const int ServerPort = 8888;
+
+        static void Main(string[] args)
+        {
+            var config = ClientConfiguration.LocalhostSilo();
+
+            GrainClient.Initialize(config);
+
+            MainAsync(args).Wait(-1); // Wait for ever, never timeout
```

Comment:
This can call ```RunServerAsync``` directly, no need for -1 in argument, that is the default if not specified.